### PR TITLE
Visualization fixes, clean-up and coverage improvements

### LINKED ISSUE #3446: Evaluate _netwrok feature not wroking when the Plotting = True, and Works when plotting= False
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Operating System

linux(EMBL BAND)

### DeepLabCut version

3.0.1

### What engine are you using?

pytorch

### DeepLabCut mode

single animal

### Device type

Nvidia Tesla V100

### Bug description 🐛

during evalation process ,plotting cant be done when kept true, but working fine when its kept False, when the multianimal feature is kept at False, and im using RESnet -50, not DLCRNET

### Steps To Reproduce

(DEEPLABCUT) bash-4.2$ python -c "
import deeplabcut
deeplabcut.evaluate_network(
'config.yaml',
Shuffles=[3],
plotting= True
)
"

### Relevant log output

```shell
(DEEPLABCUT) bash-4.2$ python -c "import deeplabcut; print(deeplabcut.__version__)"
3.0.1
(DEEPLABCUT) bash-4.2$ python -c "                                                
import deeplabcut
deeplabcut.evaluate_network(
'config.yaml',
Shuffles=[3],
plotting= False
)              
"

Evaluation scorer: DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20
100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 469/469 [00:15<00:00, 31.13it/s]
100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 53/53 [00:01<00:00, 34.13it/s]
Evaluation results file: DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20-results.csv
Evaluation results for DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20-results.csv (pcutoff: 0.6):
train rmse              6.15
train rmse_pcutoff      3.93
train mAP              98.46
train mAR              99.32
test rmse               3.97
test rmse_pcutoff       3.29
test mAP              100.00
test mAR              100.00
Name: (0.9, 3, 20, -1, 0.6), dtype: float64
(DEEPLABCUT) bash-4.2$
(DEEPLABCUT) bash-4.2$ python -c "
import deeplabcut
deeplabcut.evaluate_network(
'config.yaml',
Shuffles=[3],
plotting= True
)
"
Evaluation scorer: DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20
100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 469/469 [00:15<00:00, 31.17it/s]
100%|\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 53/53 [00:01<00:00, 35.11it/s]
Evaluation results file: DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20-results.csv
Evaluation results for DLC_Resnet50_OFTAug3shuffle3_snapshot_best-20-results.csv (pcutoff: 0.6):
train rmse              6.15
train rmse_pcutoff      3.93
train mAP              98.46
train mAR              99.32
test rmse               3.97
test rmse_pcutoff       3.29
test mAP              100.00
test mAR              100.00
Name: (0.9, 3, 20, -1, 0.6), dtype: float64
Warning: DataFrame reshape failed for img10359.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img00264.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img03469.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img13766.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img05289.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img00543.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
Warning: DataFrame reshape failed for img05787.png
  Expected: 2 individuals, 13 bodyparts
  Ground truth: 26 elements (expected 52)
  Predictions: 39 elements (expected 78)
  Skipping visualization for this image
```

### Anything else?

_No response_

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by C-Achard**
## Scope

Fixes evaluation plotting when ground-truth and prediction DataFrames use different labels, and adds coverage for the visualization utilities.

Closes #3446

## Changes

- Count individuals and body parts separately for ground truth and predictions before reshaping
- Better validate individual counts and ordered bodypart labels, skipping incompatible rows with clearer warnings
- Handle unique body parts independently for each image
- Validate plot modes early, support grayscale frame dimensions, and always close figures
- Improve visualization typing and error reporting
- Remove the duplicate module header
- Add tests for the reported regression, plot modes, malformed data, mismatches, output filenames, and bounding box styling

## Motivation

The merged DataFrame previously treated labels such as `animal` and `individual0` as two individuals, causing valid single-animal evaluation data to fail reshaping and skip visualization.

### CODE REVIEW

File: deeplabcut/utils/visualization.py [hunk 1]
```diff
@@ -515,92 +540,100 @@ def plot_evaluation_results(
...
-        elif mode == "individual":
-            colors = get_cmap(individuals + 1, name=colormap)
-        else:
-            colors = []
+        try:
+            h, w = frame.shape[:2]
+            fig.set_size_inches(w / 100, h / 100)
+            ax.set_xlim(0, w)
+            ax.set_ylim(0, h)
+            ax.invert_yaxis()
```
Comment (deruyter92) [inline deeplabcut/utils/visualization.py]: this is already inverted in `create_minimal_figure` as well. Is this intended to cancel out that inversion?

File: deeplabcut/utils/visualization.py [hunk 2]
```diff
@@ -497,12 +500,34 @@ def plot_evaluation_results(
...
+        pred_individuals = df_predictions.index.get_level_values("individuals").unique()
+
+        gt_bodyparts = df_gt.index.get_level_values("bodyparts").unique()
+        pred_bodyparts = df_predictions.index.get_level_values("bodyparts").unique()
+
+        if len(gt_individuals) != len(pred_individuals):
+            print(f"Warning: Individual count mismatch for {image}")
+            print(f"  Ground truth individual count: {len(gt_individuals)}")
+            print(f"  Predictions individual count: {len(pred_individuals)}")
+            print("  Skipping visualization for this image")
```
Comment (deruyter92) [inline deeplabcut/utils/visualization.py]: should we move to logging here instead of printing?

**Review by deruyter92 — APPROVED** [review]
Great work, definitely needed.

All changes look good, I could not find introduced regressions. Two general remarks:
- The tests are now mostly mocking the actual plotting, should we add one or two tests for `make_multi_animal_labeled_image`? If we keep it small should be fine for CI right? 
- We currently are assuming valid GT and non-empty dataframes. Should we add validation?



Suggestions for previously existing issues: 

Line 229
one remaining `h, w, numcolors = np.shape(frame)` can be changed to `h, w = np.shape(frame)[:2]` like you did elsewhere

Lines 396-398
This can be simplified with a lookup dict that is built once
```python
 for i in trange(len(ic)):
    filename = ic.files[i]
    ind = images_list.index(filename)
```

Lines 410-412
Handle the png replacement a bit more robust, e.g. :
```
stem = Path(filename).stem
out_name = f"{stem}_{color_by}.png"
```

File: deeplabcut/utils/visualization.py (hunk 2 above)
Comment (C-Achard) [reply deeplabcut/utils/visualization.py]: Yes! Thanks

File: deeplabcut/utils/visualization.py (hunk 1 above)
Comment (C-Achard) [reply deeplabcut/utils/visualization.py]: Simplified now

**Review by AlexEMG — APPROVED** [review]
